ui/src/renderer/ComponentProxy.vue

```diff
@@ -310,11 +311,25 @@ export default {
 			const vnodeProps = {
 				...getRootElProps(),
 			};
-			const renderedComponent = h(template, vnodeProps, {
-				default: defaultSlotFn,
-			});
 
-			return renderedComponent;
+			const parentId = instancePath.at(-2)?.componentId;
+			const allowedTypes = !parentId
+				? ["root"]
+				: ss.getContainableTypes(parentId);
+			if (allowedTypes.includes(component.value.type)) {
+				return h(template, vnodeProps, {
+					default: defaultSlotFn,
+				});
+			} else {
+				if (!isBeingEdited.value) return h("div");
```

Comment:
I'd do a `getRenderVNode` function as mentioned in the other component and as part of that do the

`if (!isBeingEdited.value) return h("div");`

Since it's probably something we want to do for every error type? (I know we only have this one, but I liked the `RenderError` idea)